packages/connector/src/provider.ts

```diff
@@ -81,9 +96,46 @@ export const createGianoProvider = ({ transports, chains, initialChainId, bundle
   const eventListeners: Partial<EventListeners> = {};
 
   const submitUserOperation = async (userOpRequest: any) => {
-    if (injection.onUserOperationSigned) {
-      // Hook provided: use backend validation and submission
-      return await injection.onUserOperationSigned(userOpRequest);
+    if (injection.submitUserOperation) {
+      // Hook provided: prepare complete user operation, sign it, and use backend validation and submission
+      const { calls, account, ...options } = userOpRequest;
+
+      // Prepare the user operation with gas estimates
+      const estimate = await bundler.estimateUserOperationGas({ account, calls, ...options });
+      if (!estimate) {
+        throw new Error('Could not estimate user operation');
+      }
+
+      const prepared = await bundler.prepareUserOperation({
+        account,
+        calls,
+        ...options,
+        ...estimate,
+      });
+
+      // Add default gas pricing if not provided
+      const preparedWithGas = {
+        ...prepared,
+        maxFeePerGas: options.maxFeePerGas || parseGwei('200'),
+        maxPriorityFeePerGas: options.maxPriorityFeePerGas || parseGwei('400'),
+      };
+
+      // Sign the user operation
+      const signature = await account.signUserOperation(preparedWithGas);
+
+      // Create the complete signed user operation
+      const signedUserOp = {
+        ...preparedWithGas,
+        sender: await account.getAddress(),
+        signature,
+        account: {
+          entryPoint: {
+            address: entryPoint07Address,
+          },
+        },
+      };
+
+      return await injection.submitUserOperation(signedUserOp);
     } else {
       // No hook: submit directly to bundler
       console.log({ userOpRequest });
```

Comment:
I'll be pushing commits to fix this part, missing behaviour changes from my other branch